### DISCUSSION TITLE
Need your help for renaming this plugin

### DISCUSSION THREAD

**Discussion by webdiscus** [webdiscus/html-bundler-webpack-plugin#77] [Polls]
Hello developers,

I need your help choosing a `new name` for this plugin because "html-bundler-webpack-plugin" is a _"very, very terrible name..."_, said [Andrew Lisowski](https://twitter.com/HipsterSmoothie) in the video [www.youtube.com/@devtoolsfm](https://youtu.be/w4l89214zN4?si=MbqzR7PeEvxnk70s&t=3529), and **I agree with him**.

Please, choose a best name for the plugin or write your suggestion.

**Note:** this plugin can be used as CSS extractor instead of followings:
- [mini-css-extract-plugin](https://github.com/webpack-contrib/mini-css-extract-plugin)
- [webpack-remove-empty-scripts](https://github.com/webdiscus/webpack-remove-empty-scripts)
- [resolve-url-loader](https://github.com/bholloway/resolve-url-loader)
- [svg-url-loader](https://github.com/bhovhannes/svg-url-loader)
-  see the [complete list](https://github.com/webdiscus/html-bundler-webpack-plugin#list-of-plugins)

> **NOTE**
>
> This plugin must be `Plugin number 1 for Webpack`, because with just this one plugin you can start to building your website.

The naming rules:
1. The name should contains max 3 parts, e.g., `bundler-webpack-plugin`.
2. The package name must be free in the [npm repository](https://www.npmjs.com).

**Comment by webdiscus** [comment]
> **NOTE**
>
> Please, add your variant of a plugin name here.

**Comment by davidmurdoch** [comment]
Voted `html-bundler-plugin`, but I like `bundler-plugin` just as much as `html-bundler-plugin`!

**Comment by vralle** [comment]
It's not entirely clear to me what goals will be achieved by renaming the plugin. From the obvious, a certain number of users will be disoriented because they will not be able to find the usual tool called `html-bundler-webpack-plugin`

**Reply by webdiscus** [reply]
in this video [www.youtube.com/@devtoolsfm](https://youtu.be/w4l89214zN4?si=x-eVXN-iLlOE_Gqy&t=3463) at [58:52](https://youtu.be/w4l89214zN4?si=ONTiXdt-VUJzJ1FW&t=3532): "very very terrible plugin name..., it is very difficult to find this plugin with this name".

This plugin is not popular, therefore it will be not catastrophe to rename with shortly and better name, e.g. `html-bundler` or `bundler-plugin`. 
And yet one important reason: the name `html-bundler-webpack-plugin` is confusing with the name `html-webpack-plugin`.

Many developers are really confusing and think that this plugin is an "extension" for  the very mega popular `html-webpack-plugin`.
I want to distance my plugin from the `html-webpack-plugin`.

**Reply by vralle** [reply]
I'm not sure that changing the name increases popularity. But if you do, then as I understand the purpose, the name should help identify the product. Be bright and unique. The options above contain none of this. aurora-html-plugin, webdiscus-html-plugin, fast-html-plugin. Don't know. You can ask Copilot/ChatGPT

**Reply by webdiscus** [reply]
I don't care about popularity but the name should help identify the project.
Thanks.

**Reply by vralle** [reply]
Copilot (GPT-4)
`I want to see more colorful, unique and memorable names for this plugin: https://github.com/webdiscus/html-bundler-webpack-plugin`
![html-bundler-webpack-plugin](https://github.com/webdiscus/html-bundler-webpack-plugin/assets/3823060/64d58b7b-a6bf-4114-a9f2-cc77f53dd714)

**Reply by webdiscus** [reply]
Thanks, very funny, but I will take shortly and concrete name, like `bundler-plugin`, no abstract names ;-)

**Reply by webdiscus** [reply]
I don't trust IA, only real man can do it :-)

**Reply by vralle** [reply]
`You are the real author of this plugin. And you want a new name for the plugin. What name can you come up with to better identify the plugin?`

![image](https://github.com/webdiscus/html-bundler-webpack-plugin/assets/3823060/7a10cbb4-f381-48f4-9ca7-d6e399c33168)

Maybe Copilot isn't so bad after all)

**Comment by Bessonov** [comment]
"[Everything-Everywhere-All-at-Once-webpack-plugin](https://en.wikipedia.org/wiki/Everything_Everywhere_All_at_Once)" :rofl: 

Personally, I’m fine with the current name. However, what does the plugin do at its core? I voted for "bundler-webpack-plugin," as I think it describes the main purpose of the plugin well. "entry-plugin," or perhaps "entry-webpack-plugin" (since the plugin is tied to Webpack, right?), also works.  

On the other hand, what is the long-term vision for the plugin? Do you plan to integrate support for every template engine into it? Or should the community create plugins for your plugin, making it more decentralized?  

By the way, thank you for your amazing work!

**Reply by webdiscus** [reply]
> On the other hand, what is the long-term vision for the plugin?

The plugin will be extended with new useful features. 

> Do you plan to integrate support for every template engine into it? 

Yes. All most popular template engines are already supported. Today is added support the Markdown with highlighting code.
What template engines are still missing?

> should the community create plugins for your plugin, making it more decentralized?

The plugin is extendable with custom plugins. Everybody can create own plugin using the [Plugin Hooks](https://github.com/webdiscus/html-bundler-webpack-plugin/discussions/77#plugin-hooks-and-callbacks).

**Reply by Bessonov** [reply]
> What template engines are still missing?

I don't miss any. I love Pug; however, because it is very focused on HTML, it can't be easily used for LaTeX documents or SQL scripts. Because of that, and since Handlebars takes security into account, I use Handlebars more and more. I see some efforts to make ReactJS backend-friendly. Maybe this could become my favorite template engine in the future, just because I like ReactJS and already use it for SPAs.

> Today is added support the Markdown with highlighting code.

I saw it in the commits, and it sounds great! The only danger I see is that supporting all of them can be very time-consuming.

So, based on your answers, I would still vote for bundler-webpack-plugin :+1:

**Reply by webdiscus** [reply]
> I love Pug

I love Pug template engine too. The Pug compiler is fastest.

Personally, I don't like React-like frameworks. 
React is an anti pattern, therefore I promote using simple and right way with canonical template engines.

**Reply by webdiscus** [reply]
> The only danger I see is that supporting all of them can be very time-consuming.

Yes, of course, that's the other side of the coin.

## Why did I create HTML Bundler Plugin?

The existing `html-webpack-plugin` is small with limited, crooked functionality. Many [third-party plugins](https://github.com/jantimon/html-webpack-plugin?tab=readme-ov-file#plugins) were maked to expand its capabilities. But none of them solved the problem well. All of them were essentially crutches. And this whole zoo did not solve the main problem simply and clearly - render templates from source asset files of temlates, scripts, styles, etc.

It is also dangerous that different developers will simply stop supporting and developing its plugins/loaders, as has already happened with many plugins and loaders:

- [pug-loader](https://www.npmjs.com/package/pug-loader) - last update 6 years ago
- [pug-html-loader](https://npmjs.com/package/pug-html-loader) - last update 8 years ago
- [html-webpack-inject-preload](https://github.com/principalstudio/html-webpack-inject-preload) -  last update 3 years ago
-  [html-webpack-inline-source-plugin](https://github.com/dustinjackson/html-webpack-inline-source-plugin) last update 5 years ago
-  [resolve-url-loader](https://github.com/bholloway/resolve-url-loader) last update 2 years ago
-  [webpack-subresource-integrity](https://www.npmjs.com/package/webpack-subresource-integrity) last update 2 years ago
-  [handlebars-webpack-plugin](https://github.com/sagold/handlebars-webpack-plugin) last update 4 years ago
- [nunjucks-loader](https://github.com/at0g/nunjucks-loader)  last update 6 years ago
- etc..

Isn't it dangerous if many functionally important parts are not working?
Now the `html-webpack-plugin` is a leaky trough with a bunch of old crap.

### Issues

The most important thing is not even the number of contributors, but how quickly they respond and solve issues.
Look at the long [issue list by other plugins](https://github.com/jantimon/html-webpack-plugin/issues) . Many issues have been hanging for many years.

And look at the list of [yet unresolved issues](https://github.com/webdiscus/html-bundler-webpack-plugin/issues) in the bundler plugin, as well as the list of already [solved issues](https://github.com/webdiscus/html-bundler-webpack-plugin/issues?q=is%3Aissue+is%3Aclosed), how quickly was responded and how quickly was solved.

The HTML Bundler Plugin is modern, powerful, comprehensive. A full set of necessary functions out of the box. The plugin works logically and intuitively. Many people appreciated this, for example, Mozilla uses it.

What is better, a vendor lock from many single programmers, or from one (several) but making the tool convenient to use?

There is no single correct answer here. Everyone chooses what is most convenient for them.

## P.S. 

This plugin is like my child, whom I constantly develop and fix bugs for many years. 

Grüß aus Köln ;-)
